src/SupportMessage/SupportMessage.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, ReactNode } from 'react'
+import styled, { css } from 'styled-components'
+
+import { Box } from '../Box'
+import { Icon } from '../Icon'
+import { theme } from '../theme'
+
+type BorderProps =
+  | { hasBorder?: false; borderColor?: never }
+  | { hasBorder: true; borderColor: string }
+
+export type SupportMessageProps = {
+  className?: string
+  children: ReactNode
+  type: 'info' | 'alert' | 'warning'
+  title?: string
+  hasBackground?: boolean
+} & BorderProps
+
+export const SupportMessage: FC<SupportMessageProps> = ({
+  className,
+  children,
+  type = 'info',
+  hasBorder,
+  borderColor,
+  title,
+  hasBackground = true,
+}) => (
+  <Wrapper
+    className={className}
```

Comment:
since wrapper is a div instead of manually adding class name shouldn't we just extend the type like this and spread all html div props?

````
SupportMessageProps & HTMLAttributes<HTMLDivElement> = ({
   // existing props
   ... props
   })
```

and then in the wrapper you can do

`<Wrapper //existing props {...props}>`